src/rpc/RPC.cpp

```diff
@@ -324,6 +324,11 @@ buildResponse(Context const& ctx)
     if (ctx.method == "ping")
         return boost::json::object{};
 
+    if (ctx.backend->isTooBusy())
```

Comment:
Should/can this logic go in `WorkQueue`? It'd be nice to only handle `rpcTOO_BUSY` at one point in the code. 

IMO its better here, and then we only reject w/ `rpcTOO_BUSY` in WorkQueue when the queue is full. Putting this here still prevents us from sending any extra read traffic to the database.

Reply:
We don't have access to the backend in `WorkQueue` as of now, so we would need to refactor and pass that in. I think its cleaner to just reject here.

I'm not sure if we can handle `rpcTOO_BUSY` at just one point, since there are multiple things that can cause too busy. Particularly the `DatabaseTimeout` can happen at any time during handler execution.